lib/form/html.js

```diff
@@ -29,7 +29,46 @@ const addError = (html, input, errors) => {
     });
   }
 
-  result += html;
+  result += render;
+
+  return result;
+};
+
+/*
+ * Add Required Rendered Inputs
+ *
+ * @param {string} html - Rendered HTML
+ * @param {InputType} type - Input type being rendered
+ *
+ * @returns {string} - Rendered HTML, with required indicators
+ */
+const addRequired = (html, input) => {
+  let render = html;
+  let result = '';
+
+  Object.keys(input.inputs).map(inp => {
+    if (((input.hasOwnProperty('required')) && input.required === 'save') || ((input.inputs[inp].hasOwnProperty('required')) && input.inputs[inp].required === 'save')) {
+      const id = input.inputs[inp].id;
+      let level = 'required ';
+      if (input.inputs[inp].hasOwnProperty('required')) {
+        level += `required--${input.inputs[inp].required}`;
```

Comment:
Why double up on the `required` here? `required required--save` `required required--publish`? It should just be the required state, no extra `required`